### PR TITLE
PyTorch 2.0 requires ONNX to be available to use it

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

See title. Raise an exception if ONNX is used but it's not available.

cc @justusschock @awaelchli

### CODE REVIEW

File: src/lightning/pytorch/core/module.py [hunk 1]
```diff
@@ -1299,22 +1302,23 @@ def to_onnx(self, file_path: Union[str, Path], input_sample: Optional[Any] = Non
             **kwargs: Will be passed to torch.onnx.export function.
 
         Example:
-            >>> class SimpleModel(LightningModule):
```
Comment (carmocca) [inline src/lightning/pytorch/core/module.py:1302]: Removed as doctest because it's not easy to skip based on a condition (onnx available)

Comment (awaelchli) [reply src/lightning/pytorch/core/module.py:1302]: You could convert it to a regular code example, and/or move it to the .rst doc

**Review by awaelchli — APPROVED** [review]

File: src/lightning/pytorch/core/module.py (hunk 1 above)
Comment (carmocca) [reply src/lightning/pytorch/core/module.py:1302]: Isn't the current change already a conversion to a regular code example?

**Comment by carmocca** [comment]
@Borda I merged this into the other PR and it's passing. Can we move forward with this now?

**Comment by carmocca** [comment]
Included in https://github.com/Lightning-AI/lightning/pull/16764

**Pull request closed without merging**